src/spline.rs

```diff
@@ -49,10 +55,30 @@ pub struct Segment {
     ch: f64,
 }
 
+/// An imperative description of a spline path.
 #[derive(Copy, Clone, Debug)]
-enum Element {
+pub enum Element {
+    /// The start of a spline path.
     MoveTo(Point),
+    /// A line-to operation.
+    ///
+    /// The `bool` indicates whether the operation is smooth, that is whether
+    /// or not the curve into and out of the control point maintains consistent
+    /// [curvature][smoothness]. This is only relevant if this is followed
```

Comment:
Not quite right, I think, as a smooth SplineTo + a LineTo will make the last auto point in the SplineTo continuous with the line.

Reply:
cool, I was trying to figure this out experimentally and didn't feel especially confident.

Reply:
Revisiting this, and rerunning my examples, this does not seem to be influenced by the `smooth` field on the `LineTo` operation? As in, smooth SplineTo + *any* LineTo will result in the last auto point being continuous with the line?

Reply:
Ah, I think I misunderstood the scope of the comment, it was actually carefully written the first time. Yes, smooth only affects the line when followed by splineto. I thought the comment applied to the smooth bit in general, not just the line variant.

In general, you want to think of the smooth bit as controlling the join to the next segment. A line + line join is always a corner, so the smooth bit is irrelevant. To be a bit more precise, the bit only has an effect when joined to a splineto with a corresponding auto point at that side of the segment, otherwise the control point defines the tangent.

So I think the comment can stay. Likely the best way to handle the more subtle stuff is module level documentation (ideally with visual examples). Sorry for the goose chase.